security/pkg/pki/ra/k8s_ra.go

```diff
@@ -52,27 +65,47 @@ func NewKubernetesRA(raOpts *IstioRAOptions) (*KubernetesRA, error) {
 		return nil, raerror.NewError(raerror.CAInitFail, fmt.Errorf("error processing Certificate Bundle for Kubernetes RA"))
 	}
 	istioRA := &KubernetesRA{
-		csrInterface:                 raOpts.K8sClient,
-		raOpts:                       raOpts,
-		keyCertBundle:                keyCertBundle,
+		csrInterface: raOpts.K8sClient,
+		raOpts:       raOpts,
+		// CertSignerDomain is based on CERT_SIGNER_DOMAIN env variable
 		certSignerDomain:             raOpts.CertSignerDomain,
 		caCertificatesFromMeshConfig: make(map[string]string),
 	}
+	// This must be set - even if empty.
+	istioRA.keyCertBundle = keyCertBundle
```

Comment:
Why are we setting this here rather than in `istioRA` initialization?

Reply:
Oh, I had some extra code - will move it back ( but given how long this PR has been open - I would do it in another PR).